SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -75,26 +101,24 @@ void USpatialLoadBalanceEnforcer::ProcessQueuedAclAssignmentRequests()
 
 	for (WriteAuthAssignmentRequest& Request : AclWriteAuthAssignmentRequests)
 	{
-		static const int32 WarnOnAttemptNum = 5;
-		if (Request.ProcessAttempts >= WarnOnAttemptNum)
-		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Failed to process WriteAuthAssignmentRequest with EntityID: %lld. Process attempts made: %d"), Request.EntityId, Request.ProcessAttempts);
-		}
-
-		Request.ProcessAttempts++;
-
-		// TODO - if some entities won't have the component we should detect that before queueing the request.
-		// Need to be certain it is invalid to get here before receiving the AuthIntentComponent for an entity, then we can check() on it.
-		const AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(Request.EntityId);
+ 		const AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(Request.EntityId);
 		if (AuthorityIntentComponent == nullptr)
 		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Detected entity without AuthIntent component. EntityId: %lld"), Request.EntityId);
-			continue;
+			// TODO(zoning): This is possible if the ACL update lags behind the intent update. Evaluate whether this is a serious problem.
+			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("(%s) Entity without AuthIntent component will not be processed. EntityId: %lld"), *WorkerId, Request.EntityId);
+			CompletedRequests.Add(Request.EntityId);
+			return;
 		}
 
 		const FString* OwningWorkerId = VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(AuthorityIntentComponent->VirtualWorkerId);
 		if (OwningWorkerId == nullptr)
 		{
+			static const int32 WarnOnAttemptNum = 5;
```

Comment:
Can we make this class level?

Reply:
We can, but in this case I actually think I prefer it here. Anyone else have a tiebreaking opinion?

Reply:
Went to try and find a contribution guideline which, maybe intended as a joke, is a 404 
![image](https://user-images.githubusercontent.com/9222722/69742710-1f0b6200-1135-11ea-8bc9-3545b9ea0d43.png)

Reply:
This is a png of the link.

Reply:
Actually wait. Why not just make it const and not static? Then it doesn't matter where it is.

Reply:
Sure, I can just make it const and local. I'm not sure why it was ever static.

Reply:
Wont that mean it will get set multiple times to 5?

Reply:
If it's static const it will last over the lifetime of the function and will not be reinitialized each time the function is called I think.

Reply:
If you don't take an address of it the compiler is free to replace it with a literal.

Reply:
Strictly speaking in debug mode if it's not optimised then making it static means initialisation has to be thread safe and it needs to check each time if it has happened. If it's const I think the compiler can basically optimise out the whole thing in that case too though.

Reply:
I would like to just say I really don't really what we do here but because I think this is interesting...

"free to replace" - does this mean the compiler will always will replace it? If it doesn't does that mean depending on compiler you would get a "worse" version of the code?
What is the issue with it being static?

Reply:
@MatthewSandfordImprobable  Can discuss more offline.
Although I made a mistake in the above - for the static case if it's a compile time constant the debug code probably doesn't produce anything for initialising it. In the optimised case I expect them both to look the same though. And that overhead is so tiny that I wouldn't care.
The reason I wouldn't use static is because it conveys a meaning that isn't really present here. This isn't something which is shared between invocations and static locals tend to be a thing that stick out when reading code.